src/main/java/org/black_ixx/bossshop/core/rewards/BSRewardTypeTeleport.java

```diff
@@ -0,0 +1,75 @@
+package org.black_ixx.bossshop.core.rewards;
+
+import org.black_ixx.bossshop.core.BSBuy;
+import org.black_ixx.bossshop.managers.ClassManager;
+import org.black_ixx.bossshop.managers.misc.InputReader;
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.World;
+import org.bukkit.entity.Player;
+import org.bukkit.event.inventory.ClickType;
+
+public class BSRewardTypeTeleport extends BSRewardType{
+    @Override
+    public Object createObject(Object o, boolean force_final_state) {
+        return InputReader.readString(o, true);
+    }
+
+    @Override
+    public boolean validityCheck(String item_name, Object o) {
+        String str = o.toString();
+        if (str != null&&!str.isBlank()){
+            return true;
+        }
+        ClassManager.manager.getBugFinder().severe("Was not able to create ShopItem " + item_name + "! The reward object needs no to be a empty string.");
```

Comment:
Just the grammar police in me, but this should probably say "The reward object cannot be an empty string"

Reply:
sorry my english bad